test/beacon/template/json_transformer_test.exs

```diff
@@ -0,0 +1,111 @@
+defmodule Beacon.Template.HEEx.JsonTransformerTest do
+  use ExUnit.Case, async: true
+
+  alias Beacon.Template.HEEx.JsonTransformer
+
+  test "tokenizes a complex template" do
+    tokenization = [
+      {
+        :tag_block,
+        "section",
+        [],
+        [
+          {:text, "\n  ", %{newlines: 1}},
+          {
+            :tag_block,
+            "p",
+            [],
+            [{:eex, "user.name", %{column: 6, line: 2, opt: ~c"="}}],
+            %{mode: :block}
+          },
+          {:text, "\n  ", %{newlines: 1}},
+          {
+            :eex_block,
+            "if true do",
+            [
+              {
+                [
+                  {:text, " ", %{newlines: 0}},
+                  {:tag_block, "p", [], [{:text, "this", %{newlines: 0}}], %{mode: :block}}
+                ],
+                "else"
+              },
+              {[{:tag_block, "p", [], [{:text, "that", %{newlines: 0}}], %{mode: :block}}], "end"}
+            ]
+          },
+          {:text, "\n", %{newlines: 1}}
+        ],
+        %{mode: :block}
+      },
+      {:text, "\n", %{newlines: 1}},
+      {:tag_self_close, "BeaconWeb.Components.image_set",
+       [
+         {"asset", {:expr, "@beacon_live_data[:img1]", %{column: 40, line: 5}}, %{column: 33, line: 5}},
+         {"sources", {:expr, "[\"480w\"]", %{column: 75, line: 5}}, %{column: 66, line: 5}},
+         {"width", {:string, "200px", %{delimiter: 34}}, %{column: 85, line: 5}}
+       ]}
+    ]
+
+    assert JsonTransformer.transform(tokenization) ==
+             [
+               %{
+                 attrs: %{},
+                 content: [
+                   "\n  ",
+                   %{attrs: %{}, content: [%{attrs: %{}, content: "user.name", tag: :eex}], tag: "p"},
+                   "\n  ",
+                   %{
+                     arg: "if true do",
+                     blocks: [
+                       %{content: [" ", %{attrs: %{}, content: ["this"], tag: "p"}], key: "else"},
+                       %{content: [%{attrs: %{}, content: ["that"], tag: "p"}], key: "end"}
+                     ],
+                     tag: :eex_block
```

Comment:
Maybe convert this atom to string? To define that `tag` is always a string to keep the same pattern everywhere?

Reply:
Done